### PR TITLE
Verdi support

### PR DESCRIPTION
**Pull request by priyavadan**
@ginty I need some input on how to address the example.rc file template

Other than that, verdi seems to be working well in my test environment now.

### CODE REVIEW

**Comment by priyavadan** [comment]
Just noticed that if the LD_LIBRARY_PATH is not set in the user environment, the fsdb files are not generated.

Will see if I can add code to set the path automatically before the command is run.

File: lib/origen_sim/simulator.rb [hunk 1]
```diff
@@ -341,6 +359,8 @@ def run_dir
         d
       when :synopsys
         wave_dir
+      when :veridi
```
Comment (ginty) [inline lib/origen_sim/simulator.rb]: See this typo, which is easy enough to fix, but I guess means that it is currently running frim the tmp_dir, is that OK or do you need it to be the wave_dir?

**Comment by ginty** [comment]
Hi @priyavadan, can you rework this implementation to make the Verdi support an option of the Synopsys vendor?

I'm writing the documentation for OrigenSim and it is confusing trying to talk about this when it is organized by vendor and then we have Verdi treated like a vendor when it is really just a different build option for Synopsys.

If might be as simple as just saying `verdi: true` in the configuration and then implementing the Verdi branches as `if config[:verdi]` underneath the existing `if config[:vendor] == :synopsys` branches, rather than treating `:verdi` like a vendor.

Thanks!

**Comment by priyavadan** [comment]
@ginty Okay, I will make some updates today! Thanks!

**Comment by priyavadan** [comment]
@ginty running into some issues, so going to continue debug tomorrow. 

when I am done (hopefully tomorrow) :verdi will be available as a configuration option just like dve.

Thanks & Sorry!

**Comment by ginty** [comment]
@priyavadan, no probs

**Comment by priyavadan** [comment]
hi @ginty I have updated the PR. Please let me know if any questions or concerns.

File: lib/origen_sim/commands/build.rb [hunk 2]
```diff
@@ -189,4 +189,35 @@ def _exit_fail_
   puts
   puts 'See above for what to do now to create an Origen-enabled simulation object for your particular simulator.'
   puts
+  puts '-----------------------------------------------------------'
```
Comment (ginty) [inline lib/origen_sim/commands/build.rb]: This should be added above line 182

File: templates/empty.rc [hunk 3]
```diff
@@ -0,0 +1,80 @@
+Magic 271485
```
Comment (ginty) [inline templates/empty.rc]: Is this file supposed to be checked in? I don't see empty.rc actually referenced anywhere.

Comment (priyavadan) [reply templates/empty.rc]: I believe the empty wave configuration files are referenced here https://github.com/Origen-SDK/origen_sim/blob/master/lib/origen_sim/simulator.rb#L251

Comment (priyavadan) [reply templates/empty.rc]: I thought earlier that the simulator.rb referenced it, then on second glace looks like they are referenced per vendor configuration and for synopsys, dve is the default viewer with extension tcl. 

I think it should be okay to remove this file.

**Comment by ginty** [comment]
Hi @priyavadan, sorry after looking through this in more detail I think empty.rc should be re-instated.

This is used to create an initial wave config file for the user if non exists yet, it is referenced here: https://github.com/Origen-SDK/origen_sim/blob/master/lib/origen_sim/simulator.rb#L251

And from that we can see that the `wave_config_ext` method needs to be updated for Verdi - https://github.com/Origen-SDK/origen_sim/blob/master/lib/origen_sim/simulator.rb#L260

Can you also please merge latest master into this branch and setup empty.rc to include the 10 comment registers as described in #24.

So to test this, you should delete (or temporarily move) all per-user wave configs in config/waves. Run a verdi simulation, then execute the command OrigenSim provides to view the waves.
We want to see that initial wave view having a debug group that contains the comments and other items in line with the other simulators.

Thanks!

File: templates/empty.rc [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+COMPLEX_EVENT_END
+
+
+
+; toolbar current search type
+; curSTATUS search_type
+curSTATUS ByChange
+
+
+addGroup "APB"
```
Comment (ginty) [inline templates/empty.rc]: This should not contain application specific groups like this, it should be the same as what the other simulators setups provide.

**Comment by priyavadan** [comment]
@ginty sorry about that, pushed my changes in finally!

**Pull request merged**